simplq/src/components/pages/Home/MyQueues.jsx

```diff
@@ -1,41 +1,39 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import DeleteIcon from '@material-ui/icons/Delete';
 import IconButton from '@material-ui/core/IconButton';
 import { useHistory } from 'react-router';
-import { useAuth0 } from '@auth0/auth0-react';
+import { useFetchQueues, useDeleteQueue, selectQueues } from 'store/queues';
+import { useDispatch, useSelector } from 'react-redux';
 import styles from './home.module.scss';
-import { QueueRequestFactory } from '../../../api/requestFactory';
-import useRequest from '../../../api/useRequest';
 
 export default () => {
   const history = useHistory();
-  const { requestMaker } = useRequest();
-  const [myQueues, setMyQueues] = useState([]);
-  const { isAuthenticated } = useAuth0();
+  const dispatch = useDispatch();
+  const queues = useSelector(selectQueues);
+  const fetchQueues = useCallback(useFetchQueues(), []);
+  const deleteQueue = useDeleteQueue();
 
   useEffect(() => {
-    if (isAuthenticated)
-      requestMaker(QueueRequestFactory.getMyQueues()).then((resp) => setMyQueues(resp.queues));
-  }, [requestMaker, isAuthenticated]);
-
-  if (!isAuthenticated) {
-    return null;
-  }
+    dispatch(fetchQueues());
+  }, [dispatch, fetchQueues]);
 
   const handleDelete = (e, queue) => {
     // Don't trigger parent's onClick
+    const arg = { queueId: queue.queueId };
     e.stopPropagation();
-    requestMaker(QueueRequestFactory.deleteQueue(queue.queueId)).then(() => history.push('/'));
+    dispatch(deleteQueue(arg))
```

Comment:
`arg` is the same as `payload`. From my understanding, while redux action can have a form `{ type: 'deleteQueue', id: queueId, name: queueName }`, [async thunk](https://redux-toolkit.js.org/api/createAsyncThunk#payloadcreator) can accept only one argument. This also can be called in this way obviously 😄 :

```js
// Folowing the wisdom that payload should always be a single value
const queueId = queue.queueId
dispatch(deleteQueue({ queueId }))

// or
dispatch(deleteQueue({ queueId: queue.queueId }))
```

Reply:
I'll review this tomorrow morning your time.

Reply:
yes, the change makes sense. I like to avoid references if it will be used just once:
```
dispatch(deleteQueue({ queueId: queue.queueId }))
```